### PR TITLE
feat: front-end

### PR DESCRIPTION
**Pull request by alexiscolin**

### CODE REVIEW

File: web/assets/js/components/gamefen.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { Component } from "sevejs";
+import { gsap } from "gsap";
+
+const Gamefen = class extends Component {
```
Comment (zivkovicmilos) [inline web/assets/js/components/gamefen.ts:4]: What is this used for?

File: web/assets/js/actions.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
+import { io } from "socket.io-client";
+import { saveToLocalStorage, getFromLocalStorage } from "./utils/localstorage";
+
+export default new (class {
+  constructor() {
+    //INFO: MVP -> May not be the best way to handle the gno-ts
```
Comment (zivkovicmilos) [inline web/assets/js/actions.ts:6]: This is actually the best way to go. The gno-js library is (intentionally) made with exclusive reliance on the JSON-RPC server of the node, meaning there is no state management locally (apart from something that's temporarily stored in memory) that needs to happen for transactions to be processed by the node. 

The `Provider` and `Wallet` instances can be saved in this class, as they are reused for all the functionality, and are available throughout the dApp. The constructor obviously just needs to instantiate them again, to prepare them for usage (connect to the JSON-RPC server of the node, init the Wallet)

**Review by zivkovicmilos — APPROVED** [review]
Looks good for the front-end foundation 💯 

I've left a few comments, as far as I'm concerned we can merge this out 🚀

File: web/assets/js/components/gamefen.ts (hunk 1 above)
Comment (alexiscolin) [reply web/assets/js/components/gamefen.ts:4]: [sevejs](https://github.com/alexiscolin/seve-js) is a lib I coded few years ago to manage components lifecycle in creative static websites. So in our case, it manages the calls between components-classes public methods and init/destroy them at the right time. It also manage DOM manipulation.

File: web/assets/js/actions.ts (hunk 2 above)
Comment (alexiscolin) [reply web/assets/js/actions.ts:6]: Cool that's what I thought after reading your readme document.

**Pull request merged**